loader/icd_dispatch_generated.c

```diff
@@ -4250,10 +4250,10 @@ CL_API_ENTRY void CL_API_CALL clSVMFree(
         khrFirstLayer->dispatch.clSVMFree(
             context,
             svm_pointer);
-        return;
     }
+    else
 #endif // defined(CL_ENABLE_LAYERS)
-    if (context == NULL) return;
+    if (context != NULL)
```

Comment:
This throws a warning in some compiler which breaks `-Werror` builds. `ISO C forbids ‘return’ with void function` and mandatory SO discussion here with standard quote [here](https://stackoverflow.com/q/22508978/1476661).
> **Constraints**
>
>    A return statement with an expression shall not appear in a function whose return type is void. A return statement without an expression shall only appear in a function whose return type is void.

Reply:
Seems more like a bug in the compiler, as I had to create this special case to make the code conformant to this constraint. See https://github.com/KhronosGroup/OpenCL-ICD-Loader/commit/86315870729c3bfa47e4322c60c7fac37bb73338

Reply:
Is there anything I should do here?

Reply:
Either revert this change, or update the https://github.com/KhronosGroup/OpenCL-ICD-Loader/blob/master/scripts/icd_dispatch_generated.c.mako file to generate the updated code. Either one is fine by me. What do you think @bashbaug ?

Reply:
@bashbaug @Kerilk I cloned the latest `cl.xml`, updated the generating script to conform to this specific warning the extended CI caught and now it generates what my manual change did. Note that `-Werror` still can't be enabled on Linux, as few warnings still apear, for eg:
```
[ 16%] Building C object test/layer/CMakeFiles/PrintLayer.dir/icd_print_layer.c.o
/home/runner/work/OpenCL-ICD-Loader/OpenCL-ICD-Loader/test/layer/icd_print_layer.c:21:41: warning: missing field 'clGetPlatformInfo' initializer [-Wmissing-field-initializers]
struct _cl_icd_dispatch dispatch = {NULL};
                                        ^
1 warning generated.
```
We can iron these out in separate PRs. Another interesting change was that that some texture ops moved down a few hundred lines in the loader. This must be a change in the order of API entries in the XML.

I think there's nothing more to do in this PR.

Reply:
Looks good to me. I still don't understand where this warning is coming from and why I don't see it on my own Ubuntu distribution. It will be an issue as the system layer pull request heavily relies on early return and the loader doesn't rely on `goto` that much so I would like to avoid introducing too much of them...

Reply:
Do you happen to know what combination of Compiler/Standard causes those to fail?

Reply:
I tried reproducing that warning but couldn't. Turned back `-Werror` after handling the last [Clang warning](https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/160/commits/17258a03a7b0ecff02479dd3b362829e4c7ceb8e).

Reply:
Thanks for checking, I think we can resolve this @bashbaug unless you prefer to take another course of action.